src/creates/updateIssue.ts

```diff
@@ -0,0 +1,240 @@
+import { Bundle, ZObject } from "zapier-platform-core";
+import { fetchFromLinear } from "../fetchFromLinear";
+import { omitBy, uniq } from "lodash";
+
+interface IssueUpdateResponse {
+  data?: {
+    issueUpdate: {
+      issue: {
+        id: string;
+        title: string;
+        url: string;
+        identifier: string;
+      };
+      success: boolean;
+    };
+  };
+  errors?: {
+    message: string;
+    extensions?: {
+      userPresentableMessage?: string;
+    };
+  }[];
+}
+
+interface IssueResponse {
+  data: { issue: { labelIds: string[] } };
+}
+
+const updateIssueRequest = async (z: ZObject, bundle: Bundle) => {
+  if (!bundle.inputData.issueIdToUpdate) {
+    throw new z.errors.HaltedError("You must specify the ID of the issue to update");
+  }
+  const priority = bundle.inputData.priority ? parseInt(bundle.inputData.priority) : 0;
+  const estimate = bundle.inputData.estimate ? parseInt(bundle.inputData.estimate) : undefined;
+  let labelIds: string[] | undefined = undefined;
+  if (bundle.inputData.labels && bundle.inputData.labels.length > 0) {
+    // We need to append new labels to the issue's existing label set
+    const issueQuery = `
+      query ZapierIssue($id: String!) {
+        issue(id: $id) {
+          id
+          labelIds
+        }
+      }
+    `;
+    const response = await fetchFromLinear(z, bundle, issueQuery, { id: bundle.inputData.issueIdToUpdate });
+    const data = response.json as IssueResponse;
+    const originalLabelIds = data.data.issue.labelIds;
+    labelIds = uniq([...originalLabelIds, ...bundle.inputData.labels]);
```

Comment:
Man it would be nice if we had something in the API to enable adding/removing labels atomicly without having to fetch the entire set 🤔

Reply:
Agreed